sdk/maps/azure-maps-weather/src/test/java/com/azure/maps/weather/WeatherTestBase.java

```diff
@@ -43,71 +42,91 @@
 import com.azure.maps.weather.models.WeatherAlongRouteResult;
 
-public class WeatherTestBase extends TestBase {
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class WeatherTestBase extends TestProxyTestBase {
+    private static final String SDK_NAME = "client_name";
+    private static final String SDK_VERSION = "client_version";
+
     static final String FAKE_API_KEY = "fakeKeyPlaceholder";
 
-    private final String endpoint = Configuration.getGlobalConfiguration().get("API-LEARN_ENDPOINT");
-    Duration durationTestMode;
     static InterceptorManager interceptorManagerTestBase;
 
+    Duration durationTestMode;
+
     @Override
     protected void beforeTest() {
         if (interceptorManager.isPlaybackMode()) {
             durationTestMode = Duration.ofMillis(1);
         } else {
             durationTestMode = TestUtils.DEFAULT_POLL_INTERVAL;
         }
+
         interceptorManagerTestBase = interceptorManager;
     }
-    
+
     WeatherClientBuilder getWeatherAsyncClientBuilder(HttpClient httpClient, WeatherServiceVersion serviceVersion) {
         WeatherClientBuilder builder = new WeatherClientBuilder()
+            .pipeline(getHttpPipeline(httpClient))
             .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
             .serviceVersion(serviceVersion);
-        String endpoint = getEndpoint();
-        if (getEndpoint() != null) {
-            builder.endpoint(endpoint);
-        }
-        if (getTestMode() == TestMode.RECORD) {
-            builder.addPolicy(interceptorManager.getRecordPolicy());
-        }
-        if (getTestMode() == TestMode.PLAYBACK) {
-            builder.credential(new AzureKeyCredential(FAKE_API_KEY)).httpClient(interceptorManager.getPlaybackClient());
-        } else {
-            builder.credential((new AzureKeyCredential(
-                Configuration.getGlobalConfiguration().get("SUBSCRIPTION_KEY"))));
+
+        if (interceptorManager.isPlaybackMode()) {
+            builder.endpoint("https://localhost:8080");
         }
+
         return builder;
     }
 
     HttpPipeline getHttpPipeline(HttpClient httpClient) {
-        TokenCredential credential = null;
+        httpClient = interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient;
 
-        if (!interceptorManager.isPlaybackMode()) {
-            credential = new EnvironmentCredentialBuilder().httpClient(httpClient).build();
+        if (interceptorManager.isRecordMode() || interceptorManager.isPlaybackMode()) {
+            interceptorManager.addSanitizers(
+                Collections.singletonList(
+                    new TestProxySanitizer("subscription-key", ".+", "REDACTED", TestProxySanitizerType.HEADER)));
```

Comment:
need to make sure to update this with latest core.

Reply:
Will do that once we release it next cycle :)